### PR TITLE
Feature/rejected datasets - fixes #24

### PR DESCRIPTION
**Pull request by zelima**
The custom endpoint to resubmit a dataset for approval
- ApproveRejectCOntroller refactor to be more robust and handle more requests (including resubmit)
- update appropriately plugins and mailer
- properly test controllers including resubmit endpoint 

Notice on rejected datasets page
- Give ability (buttons) to edditor to delete/resubmit dataset for approval
- Customize the approval_state validator so that itcan handle approve/reject
- tests

dashboard for an editor and rejected datasets 
- get pending datasets handles not only admin scenario
- dashboard and header HTML for counting number of pending/rejected datasets
- list both pending and rejected datasets on request for editor
- tests

image

### CODE REVIEW

File: ckanext/ed/templates/package/read.html [hunk 1]
```diff
@@ -25,7 +25,15 @@
   {% elif c.pkg_dict.approval_state  == 'rejected' %}
     {% if not h.ed_is_admin(c.user)%}
     <div class="alert alert-warning" role="alert">
-      <p>{% trans %}Dataset was rejected by administrator please see comment.{% endtrans %}</p>
+      <p>{% trans %}Dataset was rejected by administrator. Please delete or modify and resubmit for approval.{% endtrans %}</p>
+      <div style="margin-top: 10px">
+        <a class="btn btn-primary" href="/dataset-publish/{{ c.pkg_dict.id }}/resubmit">Resubmit</a>
+        <a class="btn btn-danger" href="{% url_for 'dataset.delete', id=c.pkg_dict.id %}" data-module="confirm-action" data-module-content="{{ _('Are you sure you want to delete this dataset?') }}">{% block delete_button_text %}{{ _('Delete') }}{% endblock %}</a>
```
Comment (amercader) [inline ckanext/ed/templates/package/read.html]: This URL is not created properly. For 2.8 we need to use `{% url_for controller='package', action='delete', id=c.pkg_dict.id %}`

File: ckanext/ed/templates/package/read.html [hunk 2]
```diff
@@ -25,7 +25,15 @@
   {% elif c.pkg_dict.approval_state  == 'rejected' %}
     {% if not h.ed_is_admin(c.user)%}
     <div class="alert alert-warning" role="alert">
-      <p>{% trans %}Dataset was rejected by administrator please see comment.{% endtrans %}</p>
+      <p>{% trans %}Dataset was rejected by administrator. Please delete or modify and resubmit for approval.{% endtrans %}</p>
```
Comment (amercader) [inline ckanext/ed/templates/package/read.html]: Can you update to "This dataset request was rejected by an administrator. Please update it and resubmit for approval or delete it"

File: ckanext/ed/validators.py [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
-
     state = data.pop(key, None)
 
-    # If the user is member of the organization but not an admin, force the
-    # state to be pending
+    # It's hard to to tell on which action validator is executed exectly, execpt
+    # If context does not include package -> action=package_create
+    # If context includes package and state is not None -> action=package_patch
+    # If context includes package and state in not None -> action=package_update
+    if context.get('package') and not state:
```
Comment (amercader) [inline ckanext/ed/validators.py]: The proper way to do this is to have two separate validators (which can reuse much of the logic of course) and define the ones that need to be used in creation time with `create_validators` in the [dataset schema](https://github.com/CivicActions/ckanext-ed/blob/master/ckanext/ed/schemas/dataset.json#L121). In this case the ones on `validators` will only be applied at update (and patch) time.

Comment (zelima) [reply ckanext/ed/validators.py]: Created issue for this one, rest is fixed  https://github.com/CivicActions/ckanext-ed/issues/33

**Pull request merged**